docs/BuildFromSource.md

```diff
@@ -1,427 +1,195 @@
-> :bulb: To update an existing Visual Studio installation or customize the installation path, you can use the `-InstallPath` flag (`-InstallPath "D:\Program Files\Microsoft Visual Studio\2022\Enterprise"`).
-
-```powershell
-./eng/scripts/InstallVisualStudio.ps1 [-Edition {Enterprise|Community|Professional}] [-Channel {Release|Preview}]
+git clone --recursive https://github.com/YOUR_USERNAME/aspnetcore
 ```
 
-> :bulb: To execute the setup script or other PowerShell scripts in the repo, you may need to update the execution policy on your machine.
-> You can do so by running the `Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser` command
-> in PowerShell. For more information on execution policies, you can read the [execution policy docs](https://docs.microsoft.com/powershell/module/microsoft.powershell.security/set-executionpolicy).
-
-The [global.json](/global.json) file specifies the minimum requirements needed to build using `msbuild`. The [eng/scripts/vs.17.json](/eng/scripts/vs.17.json) file provides a description of the components needed to build within Visual Studio. If you plan on developing in Visual Studio, you will need to have these components installed.
-
-> :bulb: The `InstallVisualStudio.ps1` script mentioned above reads from the `vs.17[.channel].json` file to determine what components to install.
-
-#### [Git](https://git-scm.org) on Windows
-
-If you're reading this, you probably already have Git installed to support cloning the repo as outlined in Step 1.
-
-#### [NodeJS](https://nodejs.org) on Windows
-
-Building the repo requires version 16.11.0 or newer of Node. You can find installation executables for Node at <https://nodejs.org>.
-
-#### [Yarn](https://yarnpkg.com/) on Windows
-
-NodeJS installs the Node package manager (npm) by default. This repo depends on Yarn, an alternate package manager for the Node ecosystem. You can install Yarn from the command line using the following command.
-
-```powershell
-npm install -g yarn
-```
+> :bulb: All other steps below will be against your fork of the aspnetcore repo (e.g. `YOUR_USERNAME/aspnetcore`), not the official `dotnet/aspnetcore` repo.
 
-#### Java Development Kit on Windows
+> :bulb: If you've already cloned the repo without passing the `--recursive` flag, you can fetch submodule sources at any time using:
+>
+> ```bash
+> git submodule update --init --recursive
+> ```
 
-This repo contains some Java source code that depends on an install of the JDK v11 or newer. The JDK can be installed from either:
+The experience for building the repo is slightly different based on what environment you are looking to develop in. Select one of the links below to navigate to the instructions for your environment of choice.
 
-- OpenJDK <https://jdk.java.net/>
-- Oracle's JDK <https://www.oracle.com/technetwork/java/javase/downloads/index.html>
+- [Visual Studio on Windows](#visual-studio-in-windows)
```

Comment:
nit - in/on should be aligned, I'd go with `on`

Reply:
Good point! Fixed.